### PR TITLE
fix: requests error handling

### PR DESCRIPTION
**Pull request by dkhokhlov**
- added locally built docker build-image
- fixed retries for 5xx errors
- fixed various leftover build issues
 
LOG-11814

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -21,7 +21,7 @@ pipeline {
       agent {
         node {
           label 'ec2-fleet'
-          customWorkspace "${PROJECT_NAME}-${BUILD_NUMBER}"
+          //customWorkspace "${PROJECT_NAME}-${BUILD_NUMBER}"
```
Comment (stsantilena) [inline Jenkinsfile:24]: Why leave commented?

Comment (dkhokhlov) [reply Jenkinsfile]: +1, removing.

File: logdna/logdna.py [hunk 2]
```diff
@@ -190,38 +196,84 @@ def send_request(self, data):
...
+                # isn't utf-8, we fall back to iso-8859-1 for all other
+                # encodings. (See PR #3538)
+                try:
+                    reason = response.reason.decode('utf-8')
+                except UnicodeDecodeError:
+                    reason = response.reason.decode('iso-8859-1')
             else:
+                reason = response.reason
+
+            if 200 < status_code <= 399:
```
Comment (stsantilena) [inline logdna/logdna.py:240]: So only 200 would be expected code?

**Review by stsantilena — APPROVED** [review]
Just another small clarification. Other than that LGTM.

File: logdna/logdna.py (hunk 2 above)
Comment (dkhokhlov) [reply logdna/logdna.py:240]: yes, in this specific request.

**Review by esatterwhite — APPROVED** [review]

**Pull request merged**